weld-python/weld/lazy.py

```diff
@@ -301,3 +302,83 @@ def identity(phys_value, decoder):
     Creates an identity `WeldLazy` from a `PhysicalValue` and decoder.
     """
     return WeldLazy(str(phys_value.id), [phys_value], phys_value.output_type, decoder)
+
+class WeldFunc(object):
+    """
+    The result of a function annotated with `welfunc`.
+
+    This object is effectively a function with a field `code`, which extracts the raw code
+    constructed by the annotated function without constructing a `WeldLazy`.
+
+    """
+
+    __slots__ = ["_code", "_dependencies"]
+
+    def __init__(self, code, dependencies):
+        self._code = code
+        self._dependencies = dependencies
+
+    @property
+    def code(self):
```

Comment:
You don't want to add a property for the dependencies themselves? Might be useful in tests?

Reply:
I think its okay for now, since that field is not used -- no point in adding dead code.